sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -2508,26 +2515,36 @@ pi_result piMemBufferCreate(pi_context Context, pi_mem_flags Flags, size_t Size,
   pi_result Result;
   if (DeviceIsIntegrated) {
     Result = piextUSMHostAlloc(&Ptr, Context, nullptr, Size, Alignment);
+  } else if (SingleDiscreteDevice || Context->RootDevice) {
```

Comment:
`Context->RootDevice` isn't clear here at all (although I understand it's correctness)